runtime/coverage.go

```diff
@@ -483,3 +484,42 @@ func (r *CoverageReport) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// MarshalLCOV serializes each common.Location/*LocationCoverage
+// key/value pair on the *CoverageReport.Coverage map, to the
+// LCOV format. Currently supports only line coverage, function
+// and branch coverage are not yet available.
+// Description for the LCOV file format, can be found here
+// https://github.com/linux-test-project/lcov/blob/master/man/geninfo.1#L948.
+func (r *CoverageReport) MarshalLCOV() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	for location, coverage := range r.Coverage { // nolint:maprange
+		_, err := fmt.Fprintf(buf, "TN:\nSF:%s\n", location.ID())
+		if err != nil {
+			return nil, err
+		}
+		lines := make([]int, 0)
+		for line := range coverage.LineHits { // nolint:maprange
+			lines = append(lines, line)
+		}
+		sort.Ints(lines)
```

Comment:
I wonder if you need an array of lines. We know in advance that lines range from 0 to max line of the file. 
I am not sure how code coverage works but if each line is present in the map i.e. even with 0 hits then you may find the max key, so the max line and just iterate from 0 to max line while retrieving values from the map as you ve done.

Reply:
Resolved offline :pray: The bottom line is that we need to only keep track of the executable lines of a file.